File: src/MixNetFx/FacessoData/DataClasses/AddressDetailsInfo.vb
Imports ActiveDev
Imports System.Data.SqlClient

<CLSCompliant(True)> _
    Public Class AddressDetailsInfo
    Inherits InfoItemBase

    Protected myIDAddressDetail As Integer
    Protected myIDSubsidiary As Integer
    Protected myPersonnelNo As ADDBNullable(Of Integer)
    Protected myLastName As ADDBNullable(Of String)
    Protected myMiddleName As ADDBNullable(Of String)
    Protected myFirstName As ADDBNullable(Of String)
    Protected myTitle As ADDBNullable(Of String)
    Protected myStreet As ADDBNullable(Of String)
    Protected myZip As ADDBNullable(Of String)
    Protected myCity As ADDBNullable(Of String)
    Protected myCountryCode As ADDBNullable(Of String)
    Protected myCountry As ADDBNullable(Of String)
    Protected myCompanyEmail As ADDBNullable(Of String)
    Protected myPrivateEmail As ADDBNullable(Of String)
    Protected myCompanyPhone As ADDBNullable(Of String)
    Protected myPrivatePhone As ADDBNullable(Of String)
    Protected myCompanyMobile As ADDBNullable(Of String)
    Protected myPrivateMobile As ADDBNullable(Of String)
    Protected myURL As ADDBNullable(Of String)

    Sub New()
    End Sub

    Sub New(ByVal IDAddressDetail As Integer, ByVal IDSubsidiary As Integer, _
            ByVal PersonnelNo As ADDBNullable(Of Integer), ByVal LastName As ADDBNullable(Of String), _
            ByVal MiddleName As ADDBNullable(Of String), ByVal FirstName As ADDBNullable(Of String), _
            ByVal Titel As ADDBNullable(Of String), ByVal Street As ADDBNullable(Of String), _
            ByVal Zip As ADDBNullable(Of String), ByVal City As ADDBNullable(Of String), _
            ByVal CountryCode As ADDBNullable(Of String), ByVal Country As ADDBNullable(Of String), _
            ByVal CompanyTel As ADDBNullable(Of String), ByVal CompanyEmail As ADDBNullable(Of String), ByVal PrivateTel As ADDBNullable(Of String), _
            ByVal CompanyMobile As ADDBNullable(Of String), ByVal PrivateMobile As ADDBNullable(Of String), _
            ByVal PrivateEmail As ADDBNullable(Of String), ByVal URL As ADDBNullable(Of String))
        myIDAddressDetail = IDAddressDetail
        myIDSubsidiary = IDSubsidiary
        myPersonnelNo = PersonnelNo
        myLastName = LastName
        myMiddleName = MiddleName
        myFirstName = FirstName
        myTitle = Titel
        myStreet = Street
        myZip = Zip
        myCity = City
        myCountry = Country
        myCountryCode = CountryCode
        myCompanyPhone = CompanyTel
        myPrivatePhone = PrivateTel
        myCompanyMobile = CompanyMobile
        myCompanyEmail = CompanyEmail
        myPrivateMobile = PrivateMobile
        myPrivateEmail = PrivateEmail
        myURL = URL
    End Sub

    Public Property IDAddressDetail() As Integer
        Get
            Return myIDAddressDetail
        End Get
        Set(ByVal value As Integer)
            myIDAddressDetail = value
        End Set
    End Property

    Public Property IDSubsidiary() As Integer
        Get
            Return myIDSubsidiary
        End Get
        Set(ByVal value As Integer)
            myIDSubsidiary = value
        End Set
    End Property

    Public Property PersonnelNo() As ADDBNullable(Of Integer)
        Get
            Return myPersonnelNo
        End Get
        Set(ByVal value As ADDBNullable(Of Integer))
            myPersonnelNo = value
        End Set
    End Property

    Public Property LastName() As ADDBNullable(Of String)
        Get
            Return myLastName
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myLastName = value
        End Set
    End Property

    Public Property MiddleName() As ADDBNullable(Of String)
        Get
            Return myMiddleName
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myMiddleName = value
        End Set
    End Property

    Public Property FirstName() As ADDBNullable(Of String)
        Get
            Return myFirstName
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myFirstName = value
        End Set
    End Property

    Public Property Titel() As ADDBNullable(Of String)
        Get
            Return myTitle
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myTitle = value
        End Set
    End Property

    Public Property Street() As ADDBNullable(Of String)
        Get
            Return myStreet
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myStreet = value
        End Set
    End Property

    Public Property Zip() As ADDBNullable(Of String)
        Get
            Return myZip
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myZip = value
        End Set
    End Property

    Public Property City() As ADDBNullable(Of String)
        Get
            Return myCity
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCity = value
        End Set
    End Property

    Public Property CountryCode() As ADDBNullable(Of String)
        Get
            Return myCountryCode
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCountryCode = value
        End Set
    End Property

    Public Property Country() As ADDBNullable(Of String)
        Get
            Return myCountry
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCountry = value
        End Set
    End Property

    Public Property CompanyPhone() As ADDBNullable(Of String)
        Get
            Return myCompanyPhone
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCompanyPhone = value
        End Set
    End Property

    Public Property PrivatePhone() As ADDBNullable(Of String)
        Get
            Return myPrivatePhone
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myPrivatePhone = value
        End Set
    End Property

    Public Property CompanyMobile() As ADDBNullable(Of String)
        Get
            Return myCompanyMobile
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCompanyMobile = value
        End Set
    End Property

    Public Property CompanyEmail() As ADDBNullable(Of String)
        Get
            Return myCompanyEmail
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myCompanyEmail = value
        End Set
    End Property

    Public Property PrivateMobile() As ADDBNullable(Of String)
        Get
            Return myPrivateMobile
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myPrivateMobile = value
        End Set
    End Property

    Public Property PrivateEmail() As ADDBNullable(Of String)
        Get
            Return myPrivateEmail
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myPrivateEmail = value
        End Set
    End Property

    Public Property URL() As ADDBNullable(Of String)
        Get
            Return myURL
        End Get
        Set(ByVal value As ADDBNullable(Of String))
            myURL = value
        End Set
    End Property

    Public Overrides ReadOnly Property DataID() As Integer
        Get
            Return myIDAddressDetail
        End Get
    End Property

    Public Overrides ReadOnly Property DisplayName() As String
        Get
            Return LastName
        End Get
    End Property
End Class

File: src/MixNetFx/Facesso.Data/DataClasses/AddressDetailsInfo.cs
using System.Data.SqlClient;
using ActiveDev;

namespace Facesso.Data
{
    [System.CLSCompliant(true)]
    public class AddressDetailsInfo : InfoItemBase
    {
        protected int myIDAddressDetail;
        protected int myIDSubsidiary;
        protected ADDBNullable<int> myPersonnelNo;
        protected ADDBNullable<string> myLastName;
        protected ADDBNullable<string> myMiddleName;
        protected ADDBNullable<string> myFirstName;
        protected ADDBNullable<string> myTitle;
        protected ADDBNullable<string> myStreet;
        protected ADDBNullable<string> myZip;
        protected ADDBNullable<string> myCity;
        protected ADDBNullable<string> myCountryCode;
        protected ADDBNullable<string> myCountry;
        protected ADDBNullable<string> myCompanyEmail;
        protected ADDBNullable<string> myPrivateEmail;
        protected ADDBNullable<string> myCompanyPhone;
        protected ADDBNullable<string> myPrivatePhone;
        protected ADDBNullable<string> myCompanyMobile;
        protected ADDBNullable<string> myPrivateMobile;
        protected ADDBNullable<string> myURL;

        public AddressDetailsInfo() { }

        public AddressDetailsInfo(int idAddressDetail, int idSubsidiary,
            ADDBNullable<int> personnelNo, ADDBNullable<string> lastName,
            ADDBNullable<string> middleName, ADDBNullable<string> firstName,
            ADDBNullable<string> titel, ADDBNullable<string> street,
            ADDBNullable<string> zip, ADDBNullable<string> city,
            ADDBNullable<string> countryCode, ADDBNullable<string> country,
            ADDBNullable<string> companyTel, ADDBNullable<string> companyEmail,
            ADDBNullable<string> privateTel,
            ADDBNullable<string> companyMobile, ADDBNullable<string> privateMobile,
            ADDBNullable<string> privateEmail, ADDBNullable<string> url)
        {
            myIDAddressDetail = idAddressDetail;
            myIDSubsidiary = idSubsidiary;
            myPersonnelNo = personnelNo;
            myLastName = lastName;
            myMiddleName = middleName;
            myFirstName = firstName;
            myTitle = titel;
            myStreet = street;
            myZip = zip;
            myCity = city;
            myCountry = country;
            myCountryCode = countryCode;
            myCompanyPhone = companyTel;
            myPrivatePhone = privateTel;
            myCompanyMobile = companyMobile;
            myCompanyEmail = companyEmail;
            myPrivateMobile = privateMobile;
            myPrivateEmail = privateEmail;
            myURL = url;
        }

        public int IDAddressDetail
        {
            get { return myIDAddressDetail; }
            set { myIDAddressDetail = value; }
        }

        public int IDSubsidiary
        {
            get { return myIDSubsidiary; }
            set { myIDSubsidiary = value; }
        }

        public ADDBNullable<int> PersonnelNo
        {
            get { return myPersonnelNo; }
            set { myPersonnelNo = value; }
        }

        public ADDBNullable<string> LastName
        {
            get { return myLastName; }
            set { myLastName = value; }
        }

        public ADDBNullable<string> MiddleName
        {
            get { return myMiddleName; }
            set { myMiddleName = value; }
        }

        public ADDBNullable<string> FirstName
        {
            get { return myFirstName; }
            set { myFirstName = value; }
        }

        public ADDBNullable<string> Titel
        {
            get { return myTitle; }
            set { myTitle = value; }
        }

        public ADDBNullable<string> Street
        {
            get { return myStreet; }
            set { myStreet = value; }
        }

        public ADDBNullable<string> Zip
        {
            get { return myZip; }
            set { myZip = value; }
        }

        public ADDBNullable<string> City
        {
            get { return myCity; }
            set { myCity = value; }
        }

        public ADDBNullable<string> CountryCode
        {
            get { return myCountryCode; }
            set { myCountryCode = value; }
        }

        public ADDBNullable<string> Country
        {
            get { return myCountry; }
            set { myCountry = value; }
        }

        public ADDBNullable<string> CompanyPhone
        {
            get { return myCompanyPhone; }
            set { myCompanyPhone = value; }
        }

        public ADDBNullable<string> PrivatePhone
        {
            get { return myPrivatePhone; }
            set { myPrivatePhone = value; }
        }

        public ADDBNullable<string> CompanyMobile
        {
            get { return myCompanyMobile; }
            set { myCompanyMobile = value; }
        }

        public ADDBNullable<string> CompanyEmail
        {
            get { return myCompanyEmail; }
            set { myCompanyEmail = value; }
        }

        public ADDBNullable<string> PrivateMobile
        {
            get { return myPrivateMobile; }
            set { myPrivateMobile = value; }
        }

        public ADDBNullable<string> PrivateEmail
        {
            get { return myPrivateEmail; }
            set { myPrivateEmail = value; }
        }

        public ADDBNullable<string> URL
        {
            get { return myURL; }
            set { myURL = value; }
        }

        public override int DataID => myIDAddressDetail;

        public override string DisplayName => (string)LastName;
    }
}
